File: pinyougou_parent/pinyougou_manager_web/src/main/java/com/pinyougou/manager/controller/SellerController.java
package com.pinyougou.manager.controller;
import java.util.List;
import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbSeller;
import com.pinyougou.sellergoods.service.SellerService;
import com.pinyougou.util.ObjectUtils;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/seller")
public class SellerController {

    @Reference
    private SellerService sellerService;

    /**
     * 更改用户状态
     * @param sellerId
     * @param status
     * @return
     */
    @RequestMapping("/update")
    public Result update(String sellerId,String status) {
        Result update = null;
        try {
            return update = sellerService.update(sellerId, status);
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "更改失败");
        }

    }

    /**
     * 查询所有 (包含模糊查询和分页)
     * @param seller
     * @param pageNum
     * @param pageSize
     * @return
     */
    @RequestMapping("/findPage")
    public PageResult findPage(@RequestBody(required = false) TbSeller seller, Integer pageNum, Integer pageSize){
        return sellerService.listSellerByExample(seller,pageNum, pageSize);
    }

    /**
     * 删除方法
     * @param idList
     * @return
     */
    @RequestMapping("/removeSeller")
    public Result removeSeller(String[] idList) {
        if (!ObjectUtils.notEmpty(idList)) {
            return new Result(false, "未选中要删除的条目");
        }
        try {
            return sellerService.removeSeller(idList);
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "未知的错误信息,请重试");
        }
    }

    /**
     * 修改对象
     * @param seller
     * @return
     */
    @RequestMapping("/updateSeller")
    public Result updateSeller(@RequestBody(required = false) TbSeller seller) {
        if (!ObjectUtils.notEmpty(seller)) {
            return new Result(false, "对象不存在");
        }
        try {
            return sellerService.updateSeller(seller);
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "未知的错误信息,请重试");
        }
    }

    /**
     * 根据id查询
     * @return
     */
    @RequestMapping("/findOne")
    public TbSeller findOne(String id){
        return sellerService.getSellerById(id);
	}
	
    /**
     * 新增数据
     * @param seller
     * @return
     */
    @RequestMapping("/saveSeller")
    public Result saveSeller(@RequestBody(required = false) TbSeller seller){
        if (!ObjectUtils.notEmpty(seller)) {
            return new Result(false, "对象不存在");
        }
        try {
            return sellerService.saveSeller(seller);
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "未知的错误信息,请重试");
        }
    }
    /**
     * 返回全部列表(已过时的列表查询方法)
     * @return
     */
    @RequestMapping("/findAll")
    public List<TbSeller> findAll(){
        return sellerService.findAll();
    }
}

File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/controller/sellerController.js
app.controller('sellerController', function ($scope, $controller, sellerService) {//控制器
    //引入父类控制器
    $controller('baseController', {$scope: $scope});
    /**
     * 发送查询请求,定义findPage方法(已过时)
     */
    $scope.findPage = function (pageNum, pageSize) {
        sellerService.findPage(pageNum, pageSize).success(function (data) {
					  
            $scope.rows = data.rows;
            $scope.paginationConf.totalItems = data.total;//更新总记录数
        })
    };
	
    /**
     * 搜索 (升级版findPage)
	 * 读取列表数据绑定到表单中
     */
    $scope.search = function (pageNum, pageSize) {
        sellerService.search(pageNum, pageSize, $scope.searchEntity).success(function (data) {
            $scope.rows = data.rows;
            $scope.paginationConf.totalItems = data.total;//更新总记录数
        })
    };
	
    /**
     * 定义新增方法
     */
    $scope.save = function () {
        var object = null;
        if ($scope.seller.id) {
            object = sellerService.updateSeller($scope.seller);//修改
        } else {
            object = sellerService.saveSeller($scope.seller);//增加
        }
        object.success(function (data) {
					  
            if (data.success) {
                location.href = 'shoplogin.html';
            } else {
                alert(data.message);
            }
        });
    };

    /**
     * 定义根据id查找
     */
    $scope.findOne = function (id) {
							   
        sellerService.findOne(id).success(function (data) {
            $scope.seller = data;
        })
    };
	
    /**
     * 删除方法
     * @param idList
     */
    $scope.removeSellerList = function (idList) {
        if (confirm("是否删除?")) {
            sellerService.removeSellerList(idList).success(function (data) {
                if (data.success) {
                    $scope.reloadList();
                } else {
                    alert(data.message);
                }
            });
        }
    };

    /**
     * 更改用户状态
     * @param id
     * @param status
     */
    $scope.update = function (id, status) {
        sellerService.update(id, status).success(function (data) {
            if (!data.success) {
                alert(data.message);
            } else {
                $scope.reloadList();
            }
        });
    };

});
